### PR TITLE
update etcd role to deploy etcd service on Ubuntu-16.04

### LINKED ISSUE #1682: starting etcd service fails
In `contrib/ansible`, using `deploy-cluster.sh` to setup a cluster of 3 Ubuntu 16.04.1 servers (virtual machines provisioned on hyper-v), the playbook fails at role `etcd`, at `etcd-start.yml`, at task `Enable etcd`.

The error message returned by ansible is:
```
fatal: [docker1]: FAILED! => {"changed": false, "failed": true, \
"msg": "Error when trying to enable etcd: rc=1 Failed to execute operation: No such file or directory\n"}
```

If any more info is required, pls let me know.

### PR DESCRIPTION
**Pull request by ingvagabund**
SSIA

Fixes: #1682

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1698)

### CODE REVIEW

**Comment by ingvagabund** [comment]
@valentin-krasontovitsch PTAL

**Comment by ingvagabund** [comment]
So the ansible passed through etcd role. Failed on flannel but re-running again it failed on master role. So etcd can be consired fixed. @valentin-krasontovitsch can you verify? Testing in with vagrant Ubuntu 16.04 LTS) [1].

[1] https://atlas.hashicorp.com/yk0/boxes/ubuntu-xenial

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
Ubuntu 16.04 has etcd package. Hmm, we should change the source to package manager.

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
Tested on both Ubuntu-14.04 and Ubuntu-16.06. Etcd installed, service started.

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
I should have stick with the original solution installing etcd from github-release. It seems it is completely random to have the etcd running when installing package from Ubuntu. Leaving the etcdctl set command with ``etcd cluster is unavailable or misconfigured``. Switching back to github-release based etcd installation.

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
@sttts @valentin-krasontovitsch @kovvik feel free to review

File: ansible/roles/etcd/tasks/install.yml [hunk 1]
```diff
@@ -1,26 +1,72 @@
...
+    owner=etcd
+  when: etcd_source_type == "github-release"
+
+- name: Write etcd systemd unit file
+  copy: src=etcd.service dest=/etc/systemd/system
+  notify:
+    - reload systemd
+  when: etcd_source_type == "github-release"
+
+- name: Force Ubuntu-14.04 to install etcd from github release
```
Comment (sttts) [inline ansible/roles/etcd/tasks/install.yml]: Is 14.04 worth to support?

Comment (ingvagabund) [inline ansible/roles/etcd/tasks/install.yml]: It was already supported. I have just renamed to task to be more specific on its purpose.

**Comment by kovvik** [comment]
Tested in Openstack environment, it works!

**Comment by ingvagabund** [comment]
@kovvik @sttts thanks for your time, merging.

**Pull request merged**